File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/chunk/storage/HeightmapType.java
package pize.tests.voxelgame.main.chunk.storage;

import pize.tests.voxelgame.main.block.BlockData;
import pize.tests.voxelgame.client.block.Blocks;

import java.util.function.Predicate;

public enum HeightmapType{

    HIGHEST(blockID -> blockID == Blocks.AIR.getID()),
    OPAQUE(blockID -> BlockData.getProps(blockID).isLightTranslucent()),
    SURFACE(blockID -> blockID == Blocks.AIR.getID() || blockID == Blocks.WATER.getID());
    
    
    public final Predicate<Byte> isOpaque;
    
    HeightmapType(Predicate<Byte> isOpaque){
        this.isOpaque = isOpaque;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/chunk/storage/HeightmapType.kt
package pize.tests.voxelgame.main.chunk.storage

import pize.tests.voxelgame.client.block.Blocks
import pize.tests.voxelgame.main.block.BlockData
import java.util.function.Predicate

enum class HeightmapType(val isOpaque: Predicate<Byte>) {
    HIGHEST(Predicate { blockID: Byte -> blockID == Blocks.AIR.id }),
    OPAQUE(Predicate { blockID: Byte -> BlockData.getProps(blockID)!!.isLightTranslucent }),
    SURFACE(Predicate { blockID: Byte -> blockID == Blocks.AIR.id || blockID == Blocks.WATER.id })
}
